### PR TITLE
OCPBUGS-99266: move empty CRIOCredentialProviderConfig CR to run-level 0000_10

### PR DESCRIPTION
**Pull request by sdodson**
## What

Move the empty `CRIOCredentialProviderConfig` CR from run-level `0000_05` to `0000_10` so it sorts after its CRD, and add a regression test that enforces CRD-before-CR ordering for all shipped empty resources.

## Why

The empty CR ships at `0000_05_config-operator_02_criocredentialproviderconfig.cr.yaml` while its CRD ships at `0000_10_config-operator_01_criocredentialproviderconfigs-*.crd.yaml`. Under the CVO's strict update-path ordering, the `0000_NN` run-level acts as a barrier — the CVO will not advance to `0000_10` until `0000_05` completes.

The `CRIOCredentialProviderConfig` feature gate is now enabled `inDefault()`. On an upgrade into the first release that introduces this CRD, the CVO reaches run-level `0000_05`, attempts to create the CR before the CRD exists at `0000_10`, the create fails, `0000_05` never completes, and the update **deadlocks**. The SelfManagedHA CRD variant carries no `feature-set` annotation, so this affects **Default** on self-managed-high-availability clusters — not just TechPreview.

This is the master fix for OCPBUGS-99266; the 4.22 backport is openshift/api#3010.

## How

- **Rename only** (no schema change): `git mv` the empty CR to `0000_10_config-operator_02_criocredentialproviderconfig.cr.yaml`. It now sorts after the CRD within run-level `0000_10` (CRD `operatorOrdering=01` before CR `02`), mirroring the existing `0000_10_config-operator_02_node.cr.yaml` precedent.
- **New regression test** `payload-command/ordering`: walks `payload-manifests/crds` and `payload-command/empty-resources`; for every CR whose CRD is shipped in this repo, asserts the CRD is applied first under CVO rules (bootstrap-required, lower run-level, or same run-level + earlier byte-order filename — matching `LC_ALL=C`).
- **Lint config**: broaden the existing `kubeapilinter` exclusion in `.golangci.yaml` from `payload-command/*.go` to `payload-command/`. The old path only matched files directly under `payload-command/`, so the new test in the `ordering/` subpackage was linted as if its YAML-parsing helper structs were CRD API types (missing godoc/json tags, and the `nokind` naming rule renamed the `Kind` field and broke the build). `payload-command/` is tooling, not API types, so this extends the directory's existing exemption to its subpackages.

## Test

```
$ go test ./payload-command/ordering/ -v
=== RUN   TestEmptyResourceCRDOrdering
    ordering_test.go:183: verified ordering for 15 empty CR(s) against in-repo CRDs
--- PASS: TestEmptyResourceCRDOrdering
```

Verified the test fails (flagging exactly this CR/CRD pair) when the rename is reverted.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: payload-command/ordering/ordering_test.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+func (m manifest) isBootstrapRequired() bool {
+	return strings.EqualFold(m.Metadata.Annotations["release.openshift.io/bootstrap-required"], "true")
+}
+
+// appliesBefore reports whether crd is guaranteed to be applied (and, for a CRD,
+// established) before cr under CVO ordering.
+func appliesBefore(crd, cr manifest) bool {
+	if crd.isBootstrapRequired() {
+		return true
+	}
```
Comment (everettraven) [inline payload-command/ordering/ordering_test.go:79]: Just to make sure I'm following correctly - it looks like almost every other instance falls under this bootstrap requirement so that the run-level ordering actually doesn't matter.

Makes me wonder - should the CRIOCredentialProviderConfig also be marked as bootstrap required? Maybe this was overlooked in the past?

Comment (sdodson) [reply payload-command/ordering/ordering_test.go:79]: @QiWang19 Is CRIOCredentialProviderConfig expected to work during bootstrapping? I suspect it would be?

Comment (QiWang19) [reply payload-command/ordering/ordering_test.go:79]: No, supporting CRIOCredentialProviderConfig in bootstrap is not implemented.

**Comment by sdodson** [comment]
/override ci/prow/minor-e2e-upgrade-minor
Chai-bot insists that this failure is unrelated to this change, will let this sit in main for a bit before moving forward with the release-5.0 pick just to be sure
